app/controllers/ops_controller/settings/common.rb

```diff
@@ -247,6 +240,27 @@ def pglogical_validate_subscription
 
   private
 
+  PASSWOD_MASK = '●●●●●●●●'.freeze
+
+  def find_subscription(id = nil)
+    id.nil? ? PglogicalSubscription.new : PglogicalSubscription.find(id)
+  end
+
+  def set_subscription_attributes(subscription, params)
+    params_for_connection_validation(params).each do |k, v|
+      subscription.send("#{k}=".to_sym, v)
+    end
+  end
+
+  def params_for_connection_validation(subscription_params)
+    {'host'     => subscription_params[:host],
+     'port'     => subscription_params[:port],
+     'user'     => subscription_params[:user],
+     'dbname'   => subscription_params[:dbname],
+     'password' => subscription_params['password'] == PASSWOD_MASK ? nil : subscription_params['password']
```

Comment:
my bad, keys should look the same regardless of that it is working (```ActionController::Parameters``` allow to fetch values using either ```:key``` or ```'key'```